Homework 2

AnyashaTk opened the pull request:
Сделан весь объем работ, включая все указанное для тестирования функций
Обсновной объем работ:

Расчёт метрик 
Функция принимает на вход:
1.1 Словарь с инициализированными моделями
1.2 Словарь с инициализированными метриками
1.3 Инициализированный Splitter для кросс-валидации
1.4 Количество рекомендаций для генерации (K)
Реализация обучения и валидации:
1.1 RecTools Dataset через метод construct на train взаимодействиях для каждого фолда
1.2 Обучаем модель (не забываем сделать deepcopy), рекоменуем K айтемов для каждого юзера, считаем метрики на test
1.3 Дополнительно логируем время обучения
1.4 Сохраняем метрики в отчёт "Отчет_об_обучении.csv" усреднный по фолдам

Визуальный анализ
Функция принимает на вход:
1.1 Инициализированную (и обученную!) модель
1.2 Датасет
1.3 Список отобранных user_id для просмотра
1.4 item_data - данные об айтемах, которые важно отразить для визуального анализа (напр. название)
Реализация:
1.1 Генерим рекомендации для отобранных user_id
1.2 Получаем из датасета истории просмотров юзеров
1.3 Отображаем в любом удобном виде, который позволит смотреть на историю просмотра юзера + на его рекомендации от модели

AnyashaTk committed:
Hw 2 first normal commit

AnyashaTk committed:
Hw 2 готов Расчёт метрик

AnyashaTk committed:
Hw 2 готов Визуальный анализ

AnyashaTk committed:
Hw 2 отчет об обучении

feldlime commented on metrics.ipynb:
Под ноутбуки (и тем более всякие csv лучше создать папку, и туда все класть)

Под каждое ДЗ нужно создавать отдельную ветку типа `hw2`, а не одну ветку на все

Про метрики:
1. Весь код (в т.ч. для конвертации в датафрейм стоит занести внутрь функи, чтобы потом было удобнее)
2. Порядок метрик в датафрейме стоит сделать более адекватным
3. А по-хорошему их еще бы сгруппировать (типа 1 уровень: Precision, Recall, ..., 2 уровень: 1, 5, 10, или наоборот)
4. Ну и раскраска бы не помешала, хотя бы как в экзамплах

Визуальный анализ:
1. Хорошо что предусмотрела обрезание истории, но ведь никто не гарантирует что датасет отсортирован
2. Код очень не эффекивный. Банальный пример: ты довольно дорогую операцию `dataset['item_id'].value_counts()` делаешь 20 раз для каждого юзера, хотя достаточно сделать 1 раз. И там таких вещей много
3. Как смотреть на итоговый датафрейм я не понимаю. Ты точно можешь глядя на 99 идущих подряд строк (которые даже не вывелись - хаха) сделать какие-то выводы??
4. Кстати, я не вникал, но почему 99, если у нас 3 юзера, у каждого 10 реко и не больше 10 действий в истории?

AnyashaTk committed:
Hw 2 исправлено:
- добавлена папка, спасибо, почему-то не подумала
- ветку пока просто переименовала, далее эту закрою и буду делать новые

Про метрики:
- конфертация в отчет добавлена в функцию
- изменена структура отчета, выглядит удобнее
- раскрашены тут только 2 значения, из-за большой группировки

Визуальный анализ:
- добавлена сортировка датасета(долго думала зачем то в прошлый раз убрала..)
- может чего-то не нашла, но оптимизировала код как минимум в 2-х местах
- 99 строк - был баг, исправила, теперь все выводится и впринципе анализируется
